velox/functions/prestosql/window/tests/WindowTestBase.cpp

```diff
@@ -50,6 +52,9 @@ void WindowTestBase::testWindowFunction(
   auto rowType = asRowType(input[0]->type());
   std::string columnsString = folly::join(", ", rowType->names());
 
+  // DuckDB uses a default of NULLS FIRST. But Velox uses NULLS LAST.
+  // ref. https://duckdb.org/docs/sql/query_syntax/orderby.
+  addDuckDBPragma("PRAGMA default_null_order='NULLS LAST'");
```

Comment:
Can we specify the null order in the SQL query explicitly?

Otherwise, consider using AssertQueryBuilder and add setNullOrder(<enum>) method to it.

Reply:
The SQL queries would become very long and confusing without the use of defaults as these have to appear for each ORDER clause where data could have nulls. 

I changed the code to use AssertQueryBuilder and setNull... for it.

Reply:
> The SQL queries would become very long and confusing without the use of defaults 

Is this so? Would you help me understand why this is the case?

Reply:
So something like "partition by c0 order by c1, c2, c3" becomes "partition by c0 order by c1 asc nulls last, c2 asc nulls last, c3 asc nulls last".  If we are testing another clause like  "partition by c0 order by c1 asc nulls first, c2 desc nulls last, c3 desc nulls first". Its more trouble to spot the differences while reading :

"partition by c0 order by c1 asc nulls last, c2 asc nulls last, c3 asc nulls last"
"partition by c0 order by c1 asc nulls last, c2 desc nulls last, c3 desc nulls first"

vs

"partition by c0 order by c1, c2, c3"
"partition by c0 order by c1, c2 desc, c3 desc nulls first"

Reply:
Got it. Thank you for explaining.